### PR TITLE
Added improvements in to_datetime Error reporting message - incorrect field today/now shown in error

### LINKED ISSUE #47495: BUG: to_datetime ParserError on incorrect element
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
pd.to_datetime(["today", "yesterday"])
```

### Issue Description

The error message indicates the error on the wrong item to convert.

Converting the string "today" to a datetime is no problem:
`pd.to_datetime("today")` gives the same result as `pd.to_datetime("now")`
`pd.to_datetime("yesterday")` gives 
`ParserError: Unknown string format: yesterday`

But combining the two into
`pd.to_datetime(["today", "yesterday"])`
gives an error indicating that the "today" string is unknown:
```
Traceback (most recent call last):
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2236, in objects_to_datetime64ns
    values, tz_parsed = conversion.datetime_to_datetime64(data.ravel("K"))
  File "pandas\_libs\tslibs\conversion.pyx", line 360, in pandas._libs.tslibs.conversion.datetime_to_datetime64
TypeError: Unrecognized value type: <class 'str'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\tools\datetimes.py", line 1076, in to_datetime
    result = convert_listlike(arg, format)
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\tools\datetimes.py", line 402, in _convert_listlike_datetimes
    result, tz_parsed = objects_to_datetime64ns(
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2242, in objects_to_datetime64ns
    raise err
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2224, in objects_to_datetime64ns
    result, tz_parsed = tslib.array_to_datetime(
  File "pandas\_libs\tslib.pyx", line 381, in pandas._libs.tslib.array_to_datetime
  File "pandas\_libs\tslib.pyx", line 613, in pandas._libs.tslib.array_to_datetime
  File "pandas\_libs\tslib.pyx", line 751, in pandas._libs.tslib._array_to_datetime_object
  File "pandas\_libs\tslib.pyx", line 742, in pandas._libs.tslib._array_to_datetime_object
  File "pandas\_libs\tslibs\parsing.pyx", line 281, in pandas._libs.tslibs.parsing.parse_datetime_string
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\dateutil\parser\_parser.py", line 1368, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\dateutil\parser\_parser.py", line 643, in parse
    raise ParserError("Unknown string format: %s", timestr)
dateutil.parser._parser.ParserError: Unknown string format: today
```

Obviously the "today" string is not the problem, it's the "yesterday" string

Switching the two strings around does gives the proper error:
`pd.to_datetime(["yesterday", "today"])`
gives
```
Traceback (most recent call last):
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2236, in objects_to_datetime64ns
    values, tz_parsed = conversion.datetime_to_datetime64(data.ravel("K"))
  File "pandas\_libs\tslibs\conversion.pyx", line 360, in pandas._libs.tslibs.conversion.datetime_to_datetime64
TypeError: Unrecognized value type: <class 'str'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\tools\datetimes.py", line 1076, in to_datetime
    result = convert_listlike(arg, format)
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\tools\datetimes.py", line 402, in _convert_listlike_datetimes
    result, tz_parsed = objects_to_datetime64ns(
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2242, in objects_to_datetime64ns
    raise err
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\pandas\core\arrays\datetimes.py", line 2224, in objects_to_datetime64ns
    result, tz_parsed = tslib.array_to_datetime(
  File "pandas\_libs\tslib.pyx", line 381, in pandas._libs.tslib.array_to_datetime
  File "pandas\_libs\tslib.pyx", line 613, in pandas._libs.tslib.array_to_datetime
  File "pandas\_libs\tslib.pyx", line 751, in pandas._libs.tslib._array_to_datetime_object
  File "pandas\_libs\tslib.pyx", line 742, in pandas._libs.tslib._array_to_datetime_object
  File "pandas\_libs\tslibs\parsing.pyx", line 281, in pandas._libs.tslibs.parsing.parse_datetime_string
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\dateutil\parser\_parser.py", line 1368, in parse
    return DEFAULTPARSER.parse(timestr, **kwargs)
  File "C:\Users\Matthijs.Kramer\Anaconda3\envs\test\lib\site-packages\dateutil\parser\_parser.py", line 643, in parse
    raise ParserError("Unknown string format: %s", timestr)
dateutil.parser._parser.ParserError: Unknown string format: yesterday
```

It seems whenever any of the input strings cannot be parsed, an error is thrown on the first item in the input.

### Expected Behavior

When providing an number of datetime strings to `pd.to_datetime`, not all of which can be parsed, the error message should indicate on which string the error occurred, rather than the first string in the list.

In the above example, the error message should read
`Unknown string format: yesterday` instead of `Unknown string format: today`, regardless of the order of the two input strings

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : e8093ba372f9adfe79439d90fe74b0b5b6dea9d6
python           : 3.10.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19044
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_Netherlands.1252

pandas           : 1.4.3
numpy            : 1.23.0
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 62.6.0
pip              : 22.1.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by dannyi96**
- closes #47495

Splitting changes from #47597, This PR aims to close the bug #47495

- Changes description
    -  For issue #47495-  made change in `pandas/_libs/tslib.pyx` to accept "now" and "today" values. These values are accepted in other areas of parsing (` _parse_today_now` in `pandas/_libs/tslib.pyx` )

- Example of changed error messages

    | Code       | Old Error ( on 1.4.3 ) | New Error |
    | ------------- | ------------- | ------------- |
    | pd.to_datetime(["today", "yesterday"]) | dateutil.parser._parser.ParserError: Unknown string format: today  | dateutil.parser._parser.ParserError: Unknown string format: yesterday present at position 1 |

### CODE REVIEW

**Comment by dannyi96** [comment]
hi @mroeschke, could you please review this change ?
Above change does the trick, but not sure if anything in addition needs to be considered here ( in terms of error handling, other existing usecases, etc )

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -813,11 +814,18 @@ cdef _array_to_datetime_object(
                 oresult[i] = 'NaT'
                 continue
             try:
-                oresult[i] = parse_datetime_string(val, dayfirst=dayfirst,
+                # Handling special case strings today & now
+                if val == "today":
```
Comment (mroeschke) [inline pandas/_libs/tslib.pyx:818]: I think this should be handled in  `parse_datetime_string`

File: pandas/_libs/tslibs/parsing.pyx [hunk 2]
```diff
@@ -1210,3 +1216,25 @@ cpdef str get_rule_month(str source):
...
+    date_string : str
+
+    Returns:
+    --------
+    datetime or None
+    """
+    cdef:
+        object res = None
+
+    if date_string == "now":
```
Comment (mroeschke) [inline pandas/_libs/tslibs/parsing.pyx:1236]: IMO I think this logic can directly go in `parse_datetime_string`

Also, this should probably return datetime objects, not Timestamp objects

File: pandas/_libs/tslibs/parsing.pyx [hunk 3]
```diff
@@ -298,6 +298,14 @@ def parse_datetime_string(
     if dt is not None:
         return dt
 
+    # Handling special case strings today & now
+    if date_string == "now":
+        dt = datetime.now()
+    elif date_string == "today":
+        dt = datetime.today()
+    if dt is not None:
```
Comment (mroeschke) [inline pandas/_libs/tslibs/parsing.pyx:306]: Nit: Can just add `return datetime...` in the if clauses instead.

Comment (dannyi96) [reply pandas/_libs/tslibs/parsing.pyx]: Sure @mroeschke , have updated the PR

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @dannyi96

**Pull request merged**